ClimsoftVer4/ClimsoftVer4/ucrFormDaily2.vb

```diff
@@ -0,0 +1,269 @@
+        '    End If
+
+        'Next
+
+        For Each kvp In dctLinkedControlsFilters
+            CallByName(fd2Record, kvp.Value.Value.GetField, CallType.Set, kvp.Key.GetValue)
+        Next
+
+    End Sub
+
+    Private Sub EnableDaysofMonth()
+        Dim ctrVFP As New ucrValueFlagPeriod
+        Dim lstShortMonths As New List(Of String)({4, 6, 9, 11})
+        Dim iMonthLength As Integer
+        Dim iMonth As Integer
+
+        If ucrLinkedMonth Is Nothing Then
+            iMonth = 1
+        Else
+            iMonth = ucrLinkedMonth.GetValue()
+        End If
+
+        If iMonth = 2 Then
+            If Not DateTime.IsLeapYear(ucrLinkedYear.GetValue) Then
+                iMonthLength = 28
+            Else
+                iMonthLength = 29
+            End If
+        Else
+            If lstShortMonths.Contains(iMonth) Then
+                iMonthLength = 30
+            Else
+                iMonthLength = 31
+            End If
+        End If
+
+        For Each ctrVFP In {ucrValueFlagPeriod29, ucrValueFlagPeriod30, ucrValueFlagPeriod31}
+            If ctrVFP.Tag <= iMonthLength Then
+                ctrVFP.Enabled = True
+            Else
+                ctrVFP.Enabled = False
+            End If
+        Next
+    End Sub
+
+    Public Sub SetYearAndMonthLink(ucrYearControl As ucrYearSelector, ucrMonthControl As ucrMonth)
+        ucrLinkedYear = ucrYearControl
+        ucrLinkedMonth = ucrMonthControl
+    End Sub
+
+    Public Sub AddUnitslink(strFieldName As String, ucrComboBox As ucrDataLinkCombobox)
+
+        If ucrLinkedUnits.ContainsKey(strFieldName) Then
+            ucrLinkedUnits.Add(strFieldName, ucrComboBox)
+        Else
+            MessageBox.Show("Developer error: This field is already linked.", caption:="Developer error")
+        End If
+
+        If Not lstTempFields.Contains(strFieldName) Then
+            lstTempFields.Add(strFieldName)
+            SetFields(lstTempFields)
+            PopulateControl()
+        End If
+
+        AddHandler ucrComboBox.evtValueChanged, AddressOf LinkedControls_evtValueChanged
+
+    End Sub
+
+    Public Sub Clear()
+
+        Dim ctr As Control
+        Dim ctrTotal As ucrTextBox
+        Dim ctrVFP As New ucrValueFlagPeriod
+        For Each ctr In Me.Controls
+            If TypeOf ctr Is ucrValueFlagPeriod Then
+                ctrVFP = ctr
+                ctrVFP.Clear()
+            ElseIf TypeOf ctr Is ucrTextBox Then
+                ctrTotal = ctr
+                ctrTotal.Clear()
+            End If
+        Next
+    End Sub
+
+    Public Sub checkTotal()
+        'Check total if required
+        ' Am not sure how to get this value yet
+        'If totalRequired = 1 Then
+
+        Dim elemTotal As Integer = 0
+        Dim expectedTotal As Integer
+        Dim ctr As Control
+        Dim ctrVFP As New ucrValueFlagPeriod
+
+        expectedTotal = ucrInputTotal.GetValue
```

Comment:
In this code statement 
`expectedTotal = ucrInputTotal.GetValue`
what happens if  ucrInputTotal.GetValue returns an empty string ?
I just got a "_An unhandled exception of type 'System.InvalidCastException' occurred in Microsoft.VisualBasic.dll
Additional information: Conversion from string "" to type 'Integer' is not valid_." when I failed to enter the total